### ISSUE TITLE
Heart rate for laps in workout adding form

### ISSUE THREAD

**Issue by jluttine** [Runalyze/Runalyze#2191]
Hi, and thanks for this awesome runalytics service! I'd like to make a feature request: Support adding avg heart rate for the laps when adding a new workout using the form. Currenlty, I can only avg HR for the whole workout session but not for each lap. Lap specific avg HR would be very interesting. Does this make sense?

**Comment by laufhannes** [comment]
If you upload a file with heart rate data, workout average and lap data will/can be calculated afterwards, so there's no need to manually set heart rate values for all laps. The input for a total average is kind of a fallback if no detailed heart rate profile is available. In that case, an estimate can be added.

The only use case for your request is a device that logs heart rate data (such that it can be evaluated lap-wise) but does not export it, right? From my point of view, that use case is negligible.

**Comment by jluttine** [comment]
Yep, the use case you describe is exactly my case. I have avg heart rates for each lap but no "raw" heart rate data. Do you know if it is possible (and easy) to write some data file manually so that I'd write one data point for each lap and then upload that file to runalyze and get the results I'm looking for? If so, what would be a good file format for that? Also, any other tips are most welcome.

**Comment by mipapo** [comment]
I'm not sure if it works: 
You could add that avg. heart rate to the tcx file
https://github.com/Runalyze/Runalyze/blob/master/tests/testfiles/tcx/Standard.tcx#L2747

Or more easily: Buy something that works correct and does the shit correct ;)

**Comment by jluttine** [comment]
Thanks! And yes, I'm buying (or bought already), but I just wanted to add some of my old marathon results to runalyze and I only had time + avg HR for each 5km.

**Comment by mipapo** [comment]
When they are "really" (1 year) old they should not have a very hard impact to the prognosis. 
And I don't think that we will rapidly implement this feature

**Comment by jluttine** [comment]
Yep, they're 6-7 years old. Just want to see all my results in one place and analyze them and my progress. I'll see if that tcx allows a convenient workaround. I don't want to lose those HR avgs so I either need to use some upload workaround or keep those HR avgs per 5km as a separate data "backup" on my laptop.

**Comment by laufhannes** [comment]
You could add this information as simple text in the notes field. That way, they won't be evaluated but you have them in Runalyze.

Creating a tcx-file for those activities is a bunch of work, I guess. The hrm-format may be easier: https://github.com/Runalyze/Runalyze/blob/master/tests/testfiles/hrm/12010401.hrm. You can skip second and third column in `[HRData]`, but the first column needs to contain 'all data points' (interval is defined in the `[Params]` section, defaults to `1s`).
